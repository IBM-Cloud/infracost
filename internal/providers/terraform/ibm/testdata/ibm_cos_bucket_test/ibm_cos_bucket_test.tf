terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "~> 1.12.0"
    }
  }
}

provider "ibm" {
    region = "global"
}

resource "ibm_resource_group" "cos_group" {
  name = "cos-resource-group"
}

resource "ibm_resource_instance" "cos_instance" {
  name              = "cos-instance"
  resource_group_id = ibm_resource_group.cos_group.id
  service           = "cloud-object-storage"
  plan              = "standard"
  location          = "global"
}

resource "ibm_cos_bucket" "standard-us-south" {
  bucket_name          = "a-standard-bucket-at-us-south"
  resource_instance_id = ibm_resource_instance.cos_instance.id
  storage_class        = "standard"
}
