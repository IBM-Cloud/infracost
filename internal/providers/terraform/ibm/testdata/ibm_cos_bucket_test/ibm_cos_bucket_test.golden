
 Name                                        Monthly Qty  Unit           Monthly Cost 
                                                                                      
 ibm_cos_bucket.standard-us-south                                                     
 ├─ Monthly Average Capacity                      10,000  GB                  $220.00 
 ├─ Class A requests                              10,000  1k API calls          $0.05 
 ├─ Class B requests                              10,000  10k API calls         $0.00 
 └─ Public Standard Egress (first 50000 GB)        1,000  GB                   $90.00 
 └─ Public Standard Egress (next 100000 GB)            0  GB                    $0.00 
 └─ Public Standard Egress (over 150000 GB)            0  GB                    $0.00 
                                                                                      
 OVERALL TOTAL                                                                $310.05 
──────────────────────────────────
3 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 1 was free:
  ∙ 1 x ibm_resource_group